scene/animation/animation_player.cpp

```diff
@@ -378,7 +384,18 @@ void AnimationPlayer::play_backwards(const StringName &p_name, double p_custom_b
 	play(p_name, p_custom_blend, -1, true);
 }
 
+void AnimationPlayer::play_section_with_markers_backwards(const StringName &p_name, const StringName &p_start_marker, const StringName &p_end_marker, double p_custom_blend) {
+	play_section_with_markers(p_name, p_start_marker, p_end_marker, p_custom_blend, -1, true);
+}
+
+void AnimationPlayer::play_section_backwards(const StringName &p_name, double p_start_time, double p_end_time, double p_custom_blend) {
+	play_section(p_name, p_start_time, p_end_time, -1, true);
+}
+
 void AnimationPlayer::play(const StringName &p_name, double p_custom_blend, float p_custom_scale, bool p_from_end) {
+#ifdef TOOLS_ENABLED
+	playing_section = false;
+#endif
```

Comment:
```suggestion
```

Isn't enough `get_current_section_start_time() || get_current_section_end_time()` ?

Reply:
Perhaps instead of setting section to 0 and animation length if section is not used, an illegal value `-1` could be set and the getter would return 0 and animation length like:

```cpp
double AnimationPlayer::get_current_section_start_time() const {
	if (playback.current.start_time != -1) {
        }
	return playback.current.start_time;
}
```

Reply:
> Isn't enough get_current_section_start_time() || get_current_section_end_time() ?

> Perhaps instead of setting section to 0 and animation length if section is not used, an illegal value -1 could be set and the getter would return 0 and animation length

The main issue with this approach is that, currently `play` is implemented by setting `playback.current.start_time` and `playback.current.end_time` as well (mainly because this allows both `play` and `play_section` to use the same code). Setting illegal values here would make normal playback fail.

This is also why I end up using an editor-only `playing_section` flag here.

Reply:
Then, it should work by replacing all direct use of playback.current.start_time with get_current_section_start_time().

Reply:
Sorry, to clarify:
* Currently, `play` is defined to be equivalent to `play_section_with_markers("", "")`, which then uses `get_marker_time` to translate markers into time, and the time is then used in `play_section` which actually implements playback (by setting `playback.current.start_time` and `playback.current.end_time`).
* `play` could alternatively be defined with `play_section` directly, but either way any illegal value would be directly used to define the playback section.
* `get_current_section_start_time()` and `get_current_section_end_time()` are already used extensively in `AnimationPlayer` whenever applicable. But making these two functions return 0 when using `play` instead of `play_section_with_markers` will break all of these use cases.
  * For example, `seek_internal` would clamp any seek position to zero if `get_current_section_start_time()` and `get_current_section_end_time()` suddenly started to return 0 because of illegal values. One wouldn't expect this behavior when using `play`.

Using illegal values this way would require adding a lot of hacky code (and maybe adding new fields to the `Playback` struct instead) that's IMO not any better than just using a bool variable.

Reply:
That is not a problem since -1 is already used as an invalid value anywhere in the Godot.

If the `playing_section` changes in multiple places or is managed separately from `current.start/end_time`, it would make sense to make it a separate bool, but that is not the case here. The change of `playing_section` is done in only one place as `play_section_with_markers()`, and it is always done at the same time as whether start/end_time is set to `0` or `animation_length`.

Also, if some codes are using `current.start/end_time` directly, you can replace it with `get_current_section_start/end_time()` as I mentions above, and the incorrect value will not be used there. Although, there is no line which uses `current.start/end_time` directly .

I have created a batch to change that, so check it out. There are only a few modifications and it should work fine. (Includes a few other modifications)

[diff.patch](https://github.com/user-attachments/files/16799953/diff.patch)

Reply:
Added some default arguments and documentation.
[diff.patch](https://github.com/user-attachments/files/16800362/diff.patch)